esp/services/ws_machine/ws_machineService.cpp

```diff
@@ -1445,69 +1446,61 @@ void Cws_machineEx::readStorageData(const char* response, CMachineInfoThreadPara
 
         if (buf.length() > 0)
         {
-            StringBuffer diskSpaceTitle;
-            int diskSpacePercentAvail = 0;
-            __int64 diskSpaceAvailable = 0, diskSpaceTotal = 0;
-            if (!readStorageSpace(buf.str(), diskSpaceTitle, diskSpaceAvailable, diskSpaceTotal, diskSpacePercentAvail))
+            Owned<CStorageData> sData = new CStorageData(true, false, true);
+            if (!sData->read(buf))
                 DBGLOG("Invalid storage information on %s: %s", pParam->m_machineData.getNetworkAddress(), buf.str());
-            else if ((diskSpaceTitle.length() > 0) && !excludePartition(diskSpaceTitle.str()))
+            else
             {
-                Owned<CStorageData> diskData = new CStorageData(diskSpaceTitle, diskSpaceAvailable, diskSpaceTotal, diskSpacePercentAvail);
-                storage.append(*diskData.getClear());
+                const char* title = sData->getDiskSpaceTitle();
+                if (!isEmptyString(title) && !excludePartition(title))
+                    storage.append(*sData.getClear());
             }
         }
         if (!pStr || (strnicmp(pStr, "---ProcessList1---", 18)==0))
             break;
     }
 }
 
-bool Cws_machineEx::readStorageSpace(const char *line, StringBuffer& title, __int64& free, __int64& total, int& percentAvail)
+bool CStorageData::read(const char* s)
 {
-    if (!line || !*line)
+    if (isEmptyString(s))
         return false;
 
-    StringBuffer freeStr, usedStr;
-
-    const char* pStr = line;
-    const char* pStr1 = strchr(pStr, ':');
-    if (!pStr1)
-        return false;
+    const char* pStr = s;
+    if (m_readTitle)
+    {
+        const char* ppStr = strchr(pStr, ':');
```

Comment:
Very picky, but to me the name ppStr makes me expect a "const char **".  Maybe another name?  I often just use the name "finger" for a temporary local pointer into my string.